src/data_access/DataAccessObjectFacade.java

```diff
@@ -0,0 +1,45 @@
+package data_access;
+
+import entities.Response;
+import entities.User;
+import use_case.delete.DeleteUserDataAccessInterface;
+import use_case.set_response.SetResponseDataAccessInterface;
+
+import java.util.UUID;
+
+public class DataAccessObjectFacade implements SetResponseDataAccessInterface, DeleteUserDataAccessInterface {
+    FileUserDataAccessObject userDataAccessObject;
+    public DataAccessObjectFacade(FileUserDataAccessObject userDataAccessObject) {
+        this.userDataAccessObject = userDataAccessObject;
+    }
+
+    @Override
+    public boolean responseExistsById(UUID responseId) {
+        return userDataAccessObject.responseExistsById(responseId);
+    }
+
+    @Override
+    public void deleteResponse(UUID responseId) {
+        userDataAccessObject.deleteResponse(responseId);
+    }
+
+    @Override
+    public void setResponse(Response response) {
+        userDataAccessObject.setResponse(response);
+    }
+
+    @Override
+    public String getActivePromptText() {
+        return null;
+    }
+
+    @Override
+    public UUID getActivePromptId() {
```

Comment:
Likely do not need to implement this method within the UserDAO, can use the PromptDAO for this method.

Reply:
That specific method is delegating to just the prompt DAO, not the User DAO! I just didn’t have the code for the Prompt DAO at the time so it returns null.